api/src/controllers/team.js

```diff
@@ -310,6 +315,114 @@ async function updateTeam(req, res, next) {
+  * @returns 200: teamn is deleted
+  * @returns 404: team is not found
+  * @returns 400: team id is not in a valid hexadecimal format
+  */
+async function deleteTeam(req, res, next) {
+  try {
+    const { team_id: _id } = req.params;
+
+    await HomePage.deleteOne({ teamId: _id });
+    await Website.deleteOne({ teamId: _id });
+    await Achievement.deleteOne({ teamId: _id });
+    await Publication.deleteOne({ teamId: _id });
+    await Team.findByIdAndDelete(_id);
+
+    res.status(200).json('deleted successfully');
+  } catch (error) {
+    return next(
+      fillErrorObject(500, 'Error occurred with server', [error.message]),
+    );
+  }
+}
+
+/**
+  * Clear the team Data from the database on /team/:team_id
+  * @param {} req request object, containing team id in the url
+  * @param {*} res response object, the deleted team document
+  * @returns 200: teamn is deleted
+  * @returns 404: team is not found
+  * @returns 400: team id is not in a valid hexadecimal format
+  */
+async function clearTeam(req, res, next) {
+  try {
+    const { team_id: _id } = req.params;
+
+    await HomePage.deleteOne({ teamId: _id });
+    await Website.deleteOne({ teamId: _id });
+    await Achievement.deleteOne({ teamId: _id });
+    await Publication.deleteOne({ teamId: _id });
+
+    res.status(200).json('cleared successfully');
+  } catch (error) {
+    return next(
+      fillErrorObject(500, 'Error occurred with server', [error.message]),
+    );
+  }
+}
+
+/**
+  * Delete the team from the database on /team/:team_id
+  * @param {} req request object, containing team id in the url
+  * @param {*} res response object, the deleted team document
+  * @returns 200: teamn is deleted
+  * @returns 404: team is not found
+  * @returns 400: team id is not in a valid hexadecimal format
+  */
+async function deleteGHPages(req, res, next) {
+  const { code } = req.params;
+  // Call github API to get username
+  const { data } = await axios.get('https://api.github.com/user',
+    {
+      headers: { Authorization: `token ${code}` },
+    });
+  if (data.errors) {
+    return next(
+      fillErrorObject(400, 'Validation error', [data.errors[0].detail]),
+    );
+  }
+
+  // Creating repoName
+  const ghUsername = data.login;
+  logger.info(`GitHub deploy initiated for user: ${ghUsername}`);
+  const repoName = `${ghUsername}.github.io`;
+
+  // delete repo
+
+  let deletePages;
+  try {
+    deletePages = await axios({
+      url:
+        `https://api.github.com/repos/${
+          ghUsername
+        }/${
+          repoName
+        }`,
+      method: 'delete',
+      headers: {
+        Authorization: `token ${code}`,
+        Accept: 'application/vnd.github.v3+json',
+      },
+    });
+    // result logged
+    res.status(200).json('deleted successfully');
+    if (deletePages.status === 204) {
+      logger.info(`GitHub pages deleted for user: ${ghUsername}`);
+    } else {
+      logger.info(`GitHub pages deleted for user: ${ghUsername} failed!`);
```

Comment:
shouldn't this throw an error if it failed to delete the repo?

Reply:
Yes it does and if an error is thrown there is a catch block following this throwing 500, Error occurred within the server!

Reply:
Which means it'll never hit the else block if it fails. So it's more appropriate for the logging statement to be somewhere else, also, because of the res.status(200) in L413, L414 - 415 never actually get run.